android/src/main/kotlin/com/klaviyo/klaviyo_flutter_sdk/KlaviyoFlutterSdkPlugin.kt

```
@@ -243,6 +280,40 @@ class KlaviyoFlutterSdkPlugin :
+                    // An absent channels key means "all available marketing" - the broad grant is
+                    // only reachable through the named factory on both this SDK and Dart.
+                    val subscription =
+                        if (channelsJson == null) {
+                            Subscription.allAvailableMarketing(listId, customSource)
+                        } else {
+                            Subscription(listId, parseSubscriptionChannels(channelsJson), customSource)
+                        }
+
+                    Klaviyo.createSubscription(subscription)
+                    result.success(null)
+                } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)